packages/runtime-common/realm.ts

```diff
@@ -254,6 +257,8 @@ export class Realm {
   readonly loaderTemplate: Loader;
   readonly paths: RealmPaths;
 
+  private visibilityCache: RealmVisibility | null = null;
```

Comment:
let's set it to `undefined` by default. The semantic difference between `undefined` and `null` is that `undefined` means that we have no answer yet. `null` means we have an answer and the answer is nothing.